MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/MovieDetail/MovieDetailViewController+Design.swift

```diff
@@ -0,0 +1,114 @@
+import UIKit
+
+extension MovieDetailViewController: DesignProtocol {
+    
+    func buildViews() {
+        createViews()
+        styleViews()
+        defineLayoutForViews()
+    }
+    
+    func createViews() {
+        scrollView = UIScrollView()
+        view.addSubview(scrollView)
+        
+        contentView = UIView()
+        scrollView.addSubview(contentView)
+        
+        let details: (() -> DetailTitleViewModel?) = { [weak self] in
+            guard let self = self else { return nil }
+            
+            return self.presenter.getMovieDetails(for: self.movieId)
+        }
+        titleView = DetailTitleView(getDetailsFunction: details)
+        contentView.addSubview(titleView)
+        
+        overviewTitle = UILabel()
+        contentView.addSubview(overviewTitle)
+        
+        overview = UILabel()
+        contentView.addSubview(overview)
+        
+        let cast: (() -> [CastViewModel]) = { [weak self] in
+            guard let self = self else { return [] }
+            
+            return self.presenter.getCast(for: self.movieId)
+        }
+        castView = CastView(getCast: cast)
+        contentView.addSubview(castView)
+        
+        let recommendations: (() -> [MovieViewModel]) = { [weak self] in
+            guard let self = self else { return [] }
+            
+            return self.presenter.getRecommendations(for: self.movieId)
+        }
+        recommendationsView = RecommendationsView(getRecommendations: recommendations)
+        contentView.addSubview(recommendationsView)
+    }
+    
+    func styleViews() {
+        view.backgroundColor = .white
+        
+        scrollView.contentSize = CGSize(width: UIScreen.main.bounds.width, height: 1000)
+        scrollView.showsVerticalScrollIndicator = false
+        scrollView.isUserInteractionEnabled = true
+        scrollView.isExclusiveTouch = true
+                        
+        let backButtonItem = UIBarButtonItem(title: "", style: .plain, target: nil, action: nil)
+        navigationController?.navigationBar.topItem?.backBarButtonItem = backButtonItem
+        navigationItem.titleView = UIImageView(image: UIImage(with: .navigationBarTitleImage))
+        
+        overviewTitle.text = "Overview"
+        overviewTitle.font = .heading1
+        overviewTitle.textColor = .primaryBlue
+        
+        overview.text = presenter.getOverview(for: movieId)
+        overview.font = .regularMedium
+        overview.textColor = .black
+        overview.numberOfLines = 0
+        overview.lineBreakMode = .byWordWrapping
+        overview.setLineSpacing(lineSpacing: 0, lineHeightMultiple: 1.4)
+    }
+    
+    func defineLayoutForViews() {
+        scrollView.snp.makeConstraints {
+            $0.edges.equalToSuperview()
+        }
+        
+        contentView.snp.makeConstraints {
+            $0.width.equalTo(UIScreen.main.bounds.width)
```

Comment:
Not `UIScreen.main.bounds.width` because that's static and this viewController might not be in full screen. You can use `view` instead to match the actual view controller width.

Reply:
[Here](https://github.com/lihter/Movie-App/pull/28/commits/3a1dd8418b7535a8fac5faf7eea2183f097a6eef#diff-0e250b879344a24765e175866cb828b87d12e258a723eb41dcf4344c1dcd19fdR76)